factor/token/version.php

```diff
@@ -26,9 +26,14 @@
 
 defined('MOODLE_INTERNAL') || die();
 
-$plugin->version      = 2023101600;      // The current plugin version (Date: YYYYMMDDXX).
+$plugin->version      = 2023080100;      // The current plugin version (Date: YYYYMMDDXX).
 $plugin->requires     = 2022041908; // Support Moodle 4.0 and higher.
 $plugin->component    = 'factor_token';
 $plugin->release      = 2022011700;
 $plugin->maturity     = MATURITY_STABLE;
 $plugin->dependencies = ['tool_mfa' => 2019102400];
+
+// Fix version numbers that are higher than 4.3 core.
+if (!during_initial_install() && get_config('factor_token', 'version') >= '2023100900') {
```

Comment:
This string/int comparison doesnt feel quite right. Can you verify the type and do an explicit cast if required. Having it hardcoded as a string feels wrong.

Reply:
Thanks @Peterburnett 

I copied from string part from the [example](https://github.com/catalyst/moodle-local_maillog/blob/MOODLE_35_STABLE/version.php) as get_config('factor_token', 'version') is returning a string, but that's a good point. 

This is running on every page so we'd want to go with the most efficient comparison. What do you think would be most efficient - leaving it as a string or casting with (int) or intval()?

Reply:
After some testing I ended up going with (int) get_config('factor_token', 'version') >= 2023100900 as the difference was almost negligible compared to the other check and getting the config.